controllers/storagecluster/reconcile.go

```diff
@@ -340,22 +342,29 @@ func (r *StorageClusterReconciler) reconcilePhases(
 	r.phase = ""
 	var objs []resourceManager
 	if !instance.Spec.ExternalStorage.Enable {
-		// list of default ensure functions
-		// preserve list order
-		objs = []resourceManager{
-			&ocsCephConfig{},
-			&ocsCephCluster{},
-			&ocsCephBlockPools{},
-			&ocsCephFilesystems{},
-			&ocsCephObjectStores{},
-			&ocsCephObjectStoreUsers{},
-			&ocsCephRGWRoutes{},
-			&ocsStorageClass{},
-			&ocsNoobaaSystem{},
-			&ocsStorageQuota{},
-			&ocsSnapshotClass{},
-			&ocsJobTemplates{},
-			&ocsCephRbdMirrors{},
+		if !r.IsNoobaaStandalone {
```

Comment:
It looks like this is being checked in the reconcile operation only for non-external clusters.  Is this the expected behaviour?

Reply:
Right... Currently it is like this:

* internal-mode + noobaa-standalone ==> noobaa-only
* external-mode + noobaa-standalone ==> external mode (and noobaa-standalone is ignored)

One could argue that noobaa-standalone should alway win.

Or it could be made independent of the cephcluster reconciliation, i.e. noobaa-standalone would mean that noobaa is just not using ceph for backing store. And the ceph cluster could be managed independently, internally or externally.

Reply:
Specifying both external and noobaa-only is an invalid spec to begin with. The proper thing to do would be to invalidate the spec and refuse to reconcile until it is resolved. After that the issue of which one takes precedent is largely a moot point.

As far as non-ceph noobaa goes, we can already do that by explicitly specifying the noobaa db storageclassname. That's the only thing that ocs-operator generates for noobaa from ceph. Here, the standalone reconcile strategy is shorthand for don't create any Ceph resources. Without it, the user would have to specify `reconcileStrategy: ignore` for every single other managed resource. And if we do that, then we don't need `standalone` since the absence of the RBD storageclass would cause noobaa to use some other default storageclass. Also we would continue to have the (irritating) situation where `standalone == ignore`.

Reply:
I think we can merge this is as is for the feature completion and put the rest it as fixes in subsequent PRs

Reply:
Just to record it: I agreed to let noobaa-standalone take precedence. But in exploring that, I found additional complications. So a subsequent PR should include at minimum:

1. Let NooBaa-standalone take precedence over external CephCluster
2. Check for StorageClass in NooBaa creation

Additionally for external mode, we should maybe figure out a way to pre-emptively detect if RBD StorageClass exists in external Secret. If it doesn't, StorageClass reconciliation will still succeed, so NooBaa will try to use a StorageClass that doesn't exist. Whether we fail or fall back to defaults is still up for debate.